### PR TITLE
tests: pull from urandom when real entropy is not enough

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This is needed to prevent a timeout in the create-key command due to
lack of entropy. This scenario is produced sporadically when there is a
segfault produced by rngd.

This is the error which is affecting:
Jun 10 01:39:19 ubuntu kernel: [  911.509447] rngd[3001]: segfault at
805f000 ip 0804ac3e sp bfaaa4dc error 6 in rngd[8048000+5000]

Error: 
https://travis-ci.org/snapcore/snapd/builds/241031225
More info:
https://forum.snapcraft.io/t/error-creating-key/964/6

### CODE REVIEW

File: tests/lib/entropy.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+generate_entropy() {
+	# Regenerate entropy when the level is under the entropy used by ubuntu-core (700)
+	if [ $# -gt 0 ] && [ "$1" = "--force" ] || [ $(sysctl --values kernel.random.entropy_avail || echo 0) -gt 700 ]; then
+		echo "HRNGDEVICE=/dev/urandom" > /etc/default/rng-tools
+		/etc/init.d/rng-tools restart
```
Comment (niemeyer) [inline tests/lib/entropy.sh]: This doesn't look like generating entropy?

Comment (sergiocazzolato) [inline tests/lib/entropy.sh]: @niemeyer feed_kernel_entropy_pool is better?

**Comment by sergiocazzolato** [comment]
@niemeyer the change you requested is addressed

File: tests/lib/entropy.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+feed_kernel_entropy_pool() {
+	# Feed the kernel entropy pool when the level is under 700 (level used by ubuntu-core)
+	if [ $# -gt 0 ] && [ "$1" = "--force" ] || [ $(sysctl --values kernel.random.entropy_avail || echo 0) -gt 700 ]; then
+		echo "HRNGDEVICE=/dev/urandom" > /etc/default/rng-tools
+		/etc/init.d/rng-tools restart
```
Comment (niemeyer) [inline tests/lib/entropy.sh]: I'm merging this as a step forward, but can't we do this by default in all cases during the project prepare?  We don't care about the entropy generated by those systems. Nothing we'll generate there will be used for anything but testing.

**Pull request merged**